### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by mike-sul**

### CODE REVIEW

File: src/composeappmanager.h [hunk 1]
```diff
@@ -30,15 +30,10 @@ class ComposeAppManager : public OstreeManager {
   data::InstallationResult install(const Uptane::Target& target) const override;
   std::string name() const override { return PACKAGE_MANAGER_COMPOSEAPP; };
 
- private:
-  // TODO: consider avoiding it
-  FRIEND_TEST(ComposeApp, getApps);
-  FRIEND_TEST(ComposeApp, handleRemovedApps);
-  FRIEND_TEST(ComposeApp, fetch);
```
Comment (doanac) [inline src/composeappmanager.h:118]: this is so obvious now. i don't know why i didn't do it this way to begin with :)

File: src/composeapp.cc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+bool ComposeApp::checkAvailableStorageSpace(const boost::filesystem::path& app_root, uint64_t& out_available_size) {
+  struct statvfs stat_buf {};
+  const int stat_res = statvfs(app_root.c_str(), &stat_buf);
+  if (stat_res < 0) {
+    LOG_WARNING << "Unable to read filesystem statistics: error code " << stat_res;
+    return false;
+  }
+  const uint64_t available_bytes = (static_cast<uint64_t>(stat_buf.f_bsize) * stat_buf.f_bavail);
+  const uint64_t reserved_bytes = 1 << 20;
```
Comment (doanac) [inline src/composeapp.cc:125]: this is just copied from the original source. but could you add comment like `// 1MB reserved space`

Comment (mike-sul) [reply src/composeapp.cc:125]: Sure, I forgot to do it when you asked me for the first time.

Comment (mike-sul) [reply src/composeapp.cc:127]: done

**Pull request merged**